lib/operations/tables.js

```diff
@@ -22,105 +47,175 @@ function parseColumns(columns, table, extending_type_shorthands = {}) {
-          ? `REFERENCES ${options.references}`
-          : template`REFERENCES "${options.references}"`
-      );
-      if (options.onDelete) {
-        constraints.push(`ON DELETE ${options.onDelete}`);
+  return {
+    columns: _.map(columnsWithOptions, (options, column_name) => {
+      const {
+        type,
+        collation,
+        default: defaultValue,
+        unique,
+        primaryKey,
+        notNull,
+        check,
+        references,
+      } = options;
+      const constraints = [];
+      if (collation) {
+        constraints.push(`COLLATE ${collation}`);
       }
-      if (options.onUpdate) {
-        constraints.push(`ON UPDATE ${options.onUpdate}`);
+      if (defaultValue !== undefined) {
+        constraints.push(`DEFAULT ${escapeValue(defaultValue)}`);
+      }
+      if (unique) {
+        constraints.push('UNIQUE');
+      }
+      if (primaryKey) {
+        constraints.push('PRIMARY KEY');
+      }
+      if (notNull) {
+        constraints.push('NOT NULL');
+      }
+      if (check) {
+        constraints.push(`CHECK (${check})`);
+      }
+      if (references) {
+        constraints.push(parseReferences(options));
       }
-    }
 
-    const defaultValue = options.default !== undefined ? ` DEFAULT ${escapeValue(options.default)}` : '';
-    const constraintsString = constraints.length ? ` ${constraints.join(' ')}` : '';
+      const constraintsString = constraints.length ? ` ${constraints.join(' ')}` : '';
 
-    return template`"${column_name}" ${options.type}${defaultValue}${constraintsString}`;
-  })
-    .concat(multiplePrimaryColumns
-      ? [`CONSTRAINT "${table_name}_pkey" PRIMARY KEY (${quote(primaryColumns).join(', ')})`]
-      : []
-    )
-    .join(',\n');
-}
+      return template`"${column_name}" ${type}${constraintsString}`;
+    }),
+    constraints: {
+      ...(multiplePrimaryColumns ? { primaryKey: primaryColumns } : {}),
+    },
+  };
+};
 
-export const drop = table_name =>
-  template`DROP TABLE "${table_name}";`;
+const parseConstraints = (table, options) => {
+  const {
+    check,
+    unique,
+    primaryKey,
+    foreignKeys,
+    exclude,
+    deferrable,
+    deferred,
+  } = options;
+  const table_name = typeof table === 'object' ? table.name : table;
+  const constraints = [];
+  if (check) {
+    constraints.push(`CONSTRAINT "${table_name}_chck" CHECK (${check})`);
+  }
+  if (unique) {
+    constraints.push(`CONSTRAINT "${table_name}_uniq" UNIQUE (${quote(_.isArray(unique) ? unique : [unique]).join(', ')})`);
+  }
+  if (primaryKey) {
+    constraints.push(`CONSTRAINT "${table_name}_pkey" PRIMARY KEY (${quote(_.isArray(primaryKey) ? primaryKey : [primaryKey]).join(', ')})`);
+  }
+  if (foreignKeys) {
+    (_.isArray(foreignKeys) ? foreignKeys : [foreignKeys])
+      .forEach((fk) => {
+        const {
+          columns,
+        } = fk;
+        constraints.push(`FOREIGN KEY "${_.isArray(columns) ? columns.join('", "') : columns}" ${parseReferences(fk)}`);
```

Comment:
This is producing incorrect SQL for me when I try to specify a compound foreign key. When I write

````
{
  constraints: {
    foreignKeys: [
      {
        columns: [ "columnA", "columnB" ],
        references: "otherTable (A, B)"
      }
  }
````

I expect to get `FOREIGN KEY (columnA, columnB) REFERENCES otherTable (A, B)` but instead I get `FOREIGN KEY "columnA, columnB" REFERENCES otherTable (A, B)`, which is not correct.